From 793ca11a86f523145b897954cd2a3588453ff27b Mon Sep 17 00:00:00 2001
From: "Vishnu Swaroop Kumar Sarma, Duddu" <duddu.swaroop@intel.com>
Date: Wed, 21 Aug 2019 14:33:59 +0530
Subject: [PATCH] UGW_SW-37966 pad packet to ETH_ZLEN for short pkt

pad the packet to minimum of ETH_ZLEN in xmit of ethernet driver as it leads to traffic stuck,
if a smaller packet is received
---
 drivers/net/ethernet/lantiq/ltq_eth_drv_xrx500.c | 7 +++++++
 1 file changed, 7 insertions(+)

diff --git a/drivers/net/ethernet/lantiq/ltq_eth_drv_xrx500.c b/drivers/net/ethernet/lantiq/ltq_eth_drv_xrx500.c
index bdf0157dbd97..7b840a115fbb 100644
--- a/drivers/net/ethernet/lantiq/ltq_eth_drv_xrx500.c
+++ b/drivers/net/ethernet/lantiq/ltq_eth_drv_xrx500.c
@@ -809,6 +809,13 @@ static int ltq_start_xmit(struct sk_buff *skb, struct net_device *dev)
 	int ret;
 	int len;
 
+	/* pad packet to avoid stuck in traffic queues
+	   as ethernet expects min of ETH_ZLEN bytes */
+	if (skb_put_padto(skb,ETH_ZLEN)) {
+		priv->stats.tx_dropped++;
+		return 0;
+	}
+
 	/* Call the Datapath Library's TX function */
 	((struct dma_tx_desc_1 *)&skb->DW1)->field.ep = priv->dp_subif.port_id;
 	((struct dma_tx_desc_0 *)&skb->DW0)->field.dest_sub_if_id =
